homeassistant/components/climate/__init__.py

```diff
@@ -276,128 +302,152 @@ def state_attributes(self) -> dict[str, Any]:
+            raise NotImplementedError
+        return self._attr_hvac_modes
 
     @property
     def hvac_action(self) -> str | None:
         """Return the current running hvac operation if supported.
 
         Need to be one of CURRENT_HVAC_*.
         """
-        return None
+        return self.hvac_action
 
     @property
     def current_temperature(self) -> float | None:
         """Return the current temperature."""
-        return None
+        return self._attr_current_temperature
 
     @property
     def target_temperature(self) -> float | None:
         """Return the temperature we try to reach."""
-        return None
+        return self._attr_target_temperature
 
     @property
     def target_temperature_step(self) -> float | None:
         """Return the supported step of target temperature."""
-        return None
+        return self._attr_target_temperature_step
 
     @property
     def target_temperature_high(self) -> float | None:
         """Return the highbound target temperature we try to reach.
 
         Requires SUPPORT_TARGET_TEMPERATURE_RANGE.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_target_temperature_high"):
+            raise NotImplementedError
+        return self._attr_target_temperature_high
 
     @property
     def target_temperature_low(self) -> float | None:
         """Return the lowbound target temperature we try to reach.
 
         Requires SUPPORT_TARGET_TEMPERATURE_RANGE.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_target_temperature_low"):
+            raise NotImplementedError
+        return self._attr_target_temperature_low
 
     @property
     def preset_mode(self) -> str | None:
         """Return the current preset mode, e.g., home, away, temp.
 
         Requires SUPPORT_PRESET_MODE.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_preset_mode"):
+            raise NotImplementedError
+        return self._attr_preset_mode
 
     @property
     def preset_modes(self) -> list[str] | None:
         """Return a list of available preset modes.
 
         Requires SUPPORT_PRESET_MODE.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_preset_modes"):
+            raise NotImplementedError
+        return self._attr_preset_modes
 
     @property
     def is_aux_heat(self) -> bool | None:
         """Return true if aux heater.
 
         Requires SUPPORT_AUX_HEAT.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_is_aux_heat"):
+            raise NotImplementedError
+        return self._attr_is_aux_heat
 
     @property
     def fan_mode(self) -> str | None:
         """Return the fan setting.
 
         Requires SUPPORT_FAN_MODE.
         """
-        raise NotImplementedError
+        if not hasattr(self, "_attr_fan_mode"):
+            raise NotImplementedError
+        return self._attr_fan_mode
```

Comment:
I wonder if letting it raise `AttributeError` would be fine too?
```suggestion
        return self._attr_fan_mode
```

Reply:
I think returning `None` is fine as well (like we do almost everywhere in Home Assistant).
However, I tried to keep changing current logic/behavior as much out of scope as possible.

Reply:
Well the reason we raise is such that if a user flags support for a feature, it will blow up if they don't implement the correct attributes.

Reply:
Right, from that perspective, especially with the current changes, an `AttributeError` makes more sense

Reply:
Adjusted in cases an attribute is involved.